hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/SparkFileFormatInternalRowReaderContext.scala

```diff
@@ -262,6 +265,13 @@ object SparkFileFormatInternalRowReaderContext {
     metaRefCount == filter.references.length || metaRefCount == 0
   }
 
+  /**
+   * Only valid if the filter's references only include primary key columns or {@link HoodieRecord.RECORD_KEY_METADATA_FIELD}
+   */
+  def filterIsSafeForPrimaryKey(filter: Filter, recordKeyFields: Set[String]): Boolean = {
+    filter.references.forall(c => recordKeyFields.contains(c.toLowerCase) || c.equalsIgnoreCase(HoodieRecord.RECORD_KEY_METADATA_FIELD))
```

Comment:
seems not right, all the record key fields should be included.

Reply:
> seems not right, all the record key fields should be included.

Do you mean in the case of composite primary keys? It is not necessary for all the composite primary keys to exist in the filter, because when we perform the merge, if the composite primary keys of the duplicate data used for comparison are equal, then each key in the composite primary key must also be equal.